src/games/strategy/triplea/ai/proAI/logging/ProLogWindow.java

```diff
@@ -472,7 +472,7 @@ public void run() {
   private javax.swing.JCheckBox v_enableAILogging;
   private javax.swing.JCheckBox v_limitLogHistoryCB;
   private javax.swing.JSpinner v_limitLogHistoryToSpinner;
-  private javax.swing.JComboBox v_logDepth;
+  private javax.swing.JComboBox<String> v_logDepth;
```

Comment:
Looks pretty good up to this point. Some   `@SuppressWarnings` were added that are going to bury a problem, rather than help. It's far better to have a compiler warning, than something that is actually unused, suppressed, and because it is  `@SuppressWarnings`, it'll look like it was intentionally left in because it *actually* is used, which is misleading. That leads us down a route of maintaining dead code, which is never fun to realize.